### PR TITLE
Complete dvc setup with remote and metrics

### PR DESCRIPTION
**Pull request by smrutik25**

### CODE REVIEW

**Review by mln9d4 — CHANGES_REQUESTED** [review]
Code does not work due to file path directories being written with "\\" in the src files. Should be changed to "/" and it should work in both Windows and Linux. 

After changing it locally running dvc repro works, and it trains the model. I will check if the other stages work tomorrow.

Futher more to fix the pipeline please fix the following file "REMLA_PROJECT/src/visualization/evaluation.py".

1. Re-oder import so JSON is above Seaborn
2. Breakline of the sklearn.metrics import so it doesn't become too long I.E.
`from sklearn.metrics import (classification_report,
                            confusion_matrix,
                            accuracy_score,
                            roc_auc_score,
                            f1_score)`
3. Add encoding to with open() statements : `with open(file, read/write, encoding='utf-8)`

File: REMLA_PROJECT/src/visualization/evaluation.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import pickle
 import seaborn as sns
-from sklearn.metrics import classification_report, confusion_matrix, accuracy_score
```
Comment (mln9d4) [inline REMLA_PROJECT/src/visualization/evaluation.py:3]: `import json`
should be above `import seaborn as sns`

File: REMLA_PROJECT/src/visualization/evaluation.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
-
+from sklearn.metrics import classification_report, confusion_matrix, accuracy_score, roc_auc_score, f1_score
```
Comment (mln9d4) [inline REMLA_PROJECT/src/visualization/evaluation.py:3]:
```
from sklearn.metrics import (classification_report,
                            confusion_matrix,
                            accuracy_score,
                            roc_auc_score,
                            f1_score)
```

File: REMLA_PROJECT/src/visualization/evaluation.py [hunk 3]
```diff
@@ -22,6 +22,15 @@ def evaluation():
     print(f"Accuracy: {accuracy_score(y_test, y_pred_binary)}")
     sns.heatmap(confusion_mat, annot=True)
 
+    metrics_dict = {
+    "accuracy": round(accuracy_score(y_test, y_pred_binary),5),
+    "roc_auc": round(roc_auc_score(y_test, y_pred_binary),5),
+    "f1": round(f1_score(y_test, y_pred_binary),5)
+    }
+
+    with open("REMLA_PROJECT\\reports\\metrics.json", "w") as json_file:
```
Comment (mln9d4) [inline REMLA_PROJECT/src/visualization/evaluation.py:31]: change it to `with open("REMLA_PROJECT/reports/metrics.json", "w", encoding="utf-8") as json_file:`

**Review by mln9d4 — CHANGES_REQUESTED** [review]

**Comment by smrutik25** [comment]
Made the changes and it ran successfully

**Review by mln9d4 — APPROVED** [review]

**Pull request merged**